DarkScript3/GUI.cs

```diff
@@ -627,23 +806,37 @@ public static class JSRegex
             public static Regex Comment2 = new Regex(@"(/\*.*?\*/)|(/\*.*)", RegexOptions.Singleline | RegexCompiledOption);
             public static Regex Comment3 = new Regex(@"(/\*.*?\*/)|(.*\*/)",
                                              RegexOptions.Singleline | RegexOptions.RightToLeft | RegexCompiledOption);
-            public static Regex Number = new Regex(@"\b\d+[\.]?\d*([eE]\-?\d+)?[lLdDfF]?\b|\b0x[a-fA-F\d]+\b",
+            // Allow negative sign here so that negative numbers can be highlighted as tokens.
+            public static Regex Number = new Regex(@"(\b|-)\d+[\.]?\d*([eE]\-?\d+)?[lLdDfF]?\b|\b0x[a-fA-F\d]+\b",
                                            RegexCompiledOption);
             public static Regex Keyword =
                 new Regex(
-                    @"\b(true|false|break|case|catch|const|continue|default|delete|do|else|export"
-                        + @"|for|function|if|in|instanceof|let|new|null|return|switch|this|throw|try|var|void|while|with|typeof)\b",
+                    @"\b("
+                        // Reserved in ECMAScript 2015
+                        + "break|case|catch|class|const|continue|debugger|default|delete|do|else|export|extends|finally|for|function"
+                        + "|if|import|in|instanceof|new|return|super|switch|this|throw|try|typeof|var|void|while|with|yield"
```

Comment:
can we add `from` and `as`

Reply:
I would be open to that, although they are not reserved words and still can be used as variable names. As discussed for now we can probably leave it off.